pkg/rainbow/service.go

```diff
@@ -5,43 +5,51 @@ import (
 	"time"
 )
 
-var (
-	options    = []Option{}
-	lastUpdate time.Time
-)
-
 type Service struct {
-	provider Provider
+	provider   Provider
+	lastUpdate time.Time
+	options    []Option
+	expiries   Expiries
+	tables     map[string]Table
```

Comment:
These kinds of datas are not properties of the Service, so should not be fields of the Service structure.
See #18 on how to have a interface, and then use any implementatino of the storage.